### ISSUE TITLE
spectrogram plugin: zoom with waveform

### ISSUE THREAD

**Issue by feddybear** [katspaugh/wavesurfer.js#2238]
sorry JS noob question.

I added the spectrogram plugin to a working wavesurfer object (with timeline plugin and working zoom functionality for the waveform).  The spectrogram displays correctly at 100% zoom level, but it's static.  I was expecting that the spectrogram will work plug-and-play with the pre-coded zoom logic for the waveform, but it doesn't.  Can this spectrogram zoom be done out of the box or a custom renderer is necessary?

**Comment by thijstriemstra** [comment]
I don't think the spectrogram plugin can handle zoom atm.

**Comment by feddybear** [comment]
Thanks for the response @thijstriemstra . How about if I make a custom renderer and hijack the main waveform canvases and add logic for handling spectrogram.  Do you think it's possible?  I'm getting confused with the levels of abstraction between Drawer, CanvasEntry, and MultiCanvas.  Is there some kind of guide to understand how MultiCanvas renders?

**Comment by feddybear** [comment]
Nevermind.  Found this: https://chrisparton.net/wavesurfer-js-multicanvas-renderer/. Thanks.

**Comment by zhangqing12345678** [comment]
@feddybear  
Dear, I met the same issue with you, i want the spectrogram display match with zoom value, But it only works if zoom level is 100%.
Can you give me some advices how to fix this issue? thanks!

**Comment by feddybear** [comment]
@zhangqing12345678 hi, sorry I didn't notice your message for a long time.

what I did that worked for me was to use an older version of wavesurfer's implementation of the Multicanvas renderer.  Of course it's suboptimal, as it takes a while to recompute values of frequency bins when you change zoom values, especially if the file is large.  For pre-segmented audio, it works perfectly.

I recorded a demo here (just to illustrate, but sorry I have limited capacity to instruct):
https://drive.google.com/file/d/1uAX1Ud19s2ksDDiTqXuVcEAg0Y4-quJ9/view?usp=sharing
